cmd/create/cluster/cmd.go

```diff
@@ -833,7 +833,11 @@ func run(cmd *cobra.Command, _ []string) {
 			defaultRoleARN := roleARNs[0]
 			// Prioritize roles with the default prefix
 			for _, rARN := range roleARNs {
-				if strings.Contains(rARN, fmt.Sprintf("%s-%s-Role", aws.DefaultPrefix, role.Name)) {
+				roleName, err := aws.GetResourceIdFromARN(rARN)
```

Comment:
Repeated behavior, this will be fixed in https://github.com/openshift/rosa/pull/927